### ISSUE TITLE
Responses of Type 'file' do not get deserialized correctly

### ISSUE THREAD

**Issue by baronfel** [fsprojects/SwaggerProvider#61]
### Description

This is tied logically to the previous issue I submitted, #59.  In the swaggerprovider we alias Swagger type 'file' to `byte[]`, but when I attempt to use a generated method of this type I get an exception at runtime.

The exception is because we're trying to use newtonsoft.json to `deserialize` the response string from the request into a byte[], but it's not a valid JSON value. 

### Repro steps

1. Try to use the file download endpoint from PR #60 

2. Get a serialization error at runtime

### Expected behavior

The contents of the request are returned as a byte[], because that is the representaton we've chosen for Swagger type 'File'

### Actual behavior

We observe the following serialization exception:
```
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: ÿ. Path '', line 0, position 0.
  at Newtonsoft.Json.JsonTextReader.ParseValue () [0x002b3] in <607e95f78559498690f968784b884761>:0
  at Newtonsoft.Json.JsonTextReader.Read () [0x0004c] in <607e95f78559498690f968784b884761>:0
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.ReadForType (Newtonsoft.Json.JsonReader reader, Newtonsoft.Json.Serializa
tion.JsonContract contract, System.Boolean hasConverter) [0x00003] in <607e95f78559498690f968784b884761>:0
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize (Newtonsoft.Json.JsonReader reader, System.Type objectType, S
ystem.Boolean checkAdditionalContent) [0x00029] in <607e95f78559498690f968784b884761>:0
Stopped due to error
```

or something very much like it

### Known workarounds

No workaround is possible

### Related information 

* OSX El Capitan
* latest nuget release
* Mono 4.6.2

**Comment by baronfel** [comment]
I'd be willing to look at this one, as well as the previous issue.

I think the approach should be to check the response swagger type before calling Http.RequestString, and either following that branch to return a serialized object representation if the type is not 'file' or splitting and returning either the stream or a byte[] of the stream if the swagger type is 'file'.

I'd like to look into using `Stream` instead of `byte[]` for file as well.

**Comment by sergey-tihon** [comment]
@baronfel Thank you for doing this
Ping me then PR will be ready for merge or you need my participation.

**Comment by sergey-tihon** [comment]
@baronfel Could you please describe me a benefit of using `Stream` instead of `byte[]` in this case?

**Comment by baronfel** [comment]
Well, I think we'll need a more complex type than just byte because for form data we also need to be able to specify the mimetype of the uploaded data. That could be as a byte array with an optional mimetype parameter, but I also want to go to stream so that we don't have to buffer potential totally large files into memory before writing them somewhere.

As an example, I was using the swaggerprovider to download/upload a bunch of images in parallel, and when I was downloading I'd end up loading 50ish 6MB files into memory simultaneously before being able to write them to a file.  It would be very convenient to get a stream instead that I could pipe into File IO functions.

**Comment by sergey-tihon** [comment]
Should be addressed by #131 in OpenApiClientProvider
